### PR TITLE
Add `Text` component and typography support.

### PR DESCRIPTION
**Pull request by milesj**
Hello @fabien-ml  👋 

Great library. I started building a headless UI a few months ago but yours is way more polished, so I opted to use yours instead. One thing I feel that's missing though is typography support.

This PR adds a new component called `Text` which users can use to create typography based components. Stuff like:

```tsx
<Text size="md">This is body content!</Text>
<Heading level={3}>This is level 3 heading!</Heading>
```

The special functionality of the `Text` component is that it provides a context that declares "children are rendered within a typography aware boundary". This allows components like `Link` to _contextually inherit parent styles_ instead of having to declare its own props/styles. For example:

```tsx
// Meh
<Text bold italic>
  This is some text <Link bold italic>with a link</Link> in the content.
</Text>

// Yah
<Text bold italic>
  This is some text <Link>with a link</Link> in the content.
</Text>
```

This functionality is based on Braid, which is a very well crafted design system. https://seek-oss.github.io/braid-design-system/components/Text

If you're interested in this functionality, I'll polish up the PR and write docs, otherwise feel free to close.

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -134,7 +134,7 @@ importers:
     devDependencies:
```
Comment (milesj) [inline pnpm-lock.yaml:134]: I didn't add anything here, simply running `pnpm install` updated it.

**Comment by fabien-ml** [comment]
Hi, thanks for the PR. 

However I don't understand the benefit of this instead of using the cascading nature of CSS. 

I mean, here the link already inherit the styles from p.

```html
<p class="bold">
  <Link>The link is bold</Link>
</p>
```

**Comment by milesj** [comment]
@fabien-ml This provides a mechanism that requires a Link (or other component) to be rendered within a typography boundary. Otherwise it may not be inheriting the correct styles.

```tsx
<div>
  <Link>What is this inheriting?</Link>
</div>
```

**Comment by milesj** [comment]
This pattern worked in React but doesn't actually seem to work well in Solid. Gonna close.

**Pull request closed without merging**